modules/core/src/main/java/org/locationtech/jts/geom/CoordinateSequence.java

```diff
@@ -111,15 +165,49 @@
    * @return the value of the Y ordinate in the index'th coordinate
    */
   double getY(int index);
+  
+  /**
+   * Returns ordinate Z of the specified coordinate if available.
+   * 
+   * @param index
+   * @return the value of the Z ordinate in the index'th coordinate, or Double.NaN if not defined.
+   */
+  default double getZ(int index)
+  {
+    if (hasZ()) {
+        return getOrdinate(index, 2);
+    } else {
+        return Double.NaN;
+    }
+  }
 
+  /**
+   * Returns ordinate M of the specified coordinate if available.
+   * 
+   * @param index
+   * @return the value of the Z ordinate in the index'th coordinate, or Double.NaN if not defined.
+   */
+  default double getM(int index)
+  {
+    if (hasM()) {
+      final int mIndex = getDimension()-getMeasures();
+      return getOrdinate( index, mIndex );
+    }
+    else {
+        return Double.NaN;
+    }
+  }
+  
   /**
    * Returns the ordinate of a coordinate in this sequence.
    * Ordinate indices 0 and 1 are assumed to be X and Y.
+   * <p>
    * Ordinates indices greater than 1 have user-defined semantics
-   * (for instance, they may contain other dimensions or measure values).
+   * (for instance, they may contain other dimensions or measure
+   * values as described by {@link #getDimension()} and {@link #getMeasures()}).
    *
    * @param index  the coordinate index in the sequence
-   * @param ordinateIndex the ordinate index in the coordinate (in range [0, dimension-1])
+   * @param ordinateIndex the ordinate index in the coordinate (in range [0, dimension-numberOfMeasures-1])
```

Comment:
I'm not sure about the range here.  It seems to me that the range would still be [0, dimension-1].  

The bit from [0, dimension-numberOfMeasures-1] would be the XY(Z) values.  For numberOfMeasures > 0, the bit from [dimension-numberOfMeasures, dimension-1] would the measures.

Reply:
Should you update this line of documentation then?